server/worker/worker.go

```diff
@@ -0,0 +1,338 @@
+package worker
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+type Client struct {
+	AnalyticsUrl string
+	GrafanaUrl   string
+	Token        string
+	Logger       log.Logger
+}
+
+type Dashboard struct {
+	Uid   string
+	Data  map[string]interface{}
+	Title string
+}
+
+type DashboardsResponse struct {
+	Uid string `json:"uid"`
+}
+
+type DashboardResponse struct {
+	Dashboard map[string]interface{} `json:"dashboard"`
+}
+
+type DashboardUpdateResponse struct {
+	FolderUid string `json:"folderUid"`
+	Id        uint64 `json:"id"`
+	Slug      string `json:"slug"`
+	Status    string `json:"status"`
+	Uid       string `json:"uid"`
+	Url       string `json:"url"`
+	Version   uint64 `json:"version"`
+}
+
+func getTypedDashboardData(rawDashboardData *Dashboard) map[string]interface{} {
+	dashboard, ok := rawDashboardData.Data["dashboard"]
+	var emptyDashboardData map[string]interface{}
+	if !ok {
+		return emptyDashboardData
+	}
+
+	dashboardData, ok := dashboard.(map[string]interface{})
+	if !ok {
+		return emptyDashboardData
+	}
+
+	return dashboardData
+}
+
+func getTypedPanelsData(dashboardData map[string]interface{}) []interface{} {
+	var emptyResponse []interface{}
+
+	// Dashboards data might contain folders or other nonusable "dashboards"
+	// All usable dashboards have minimum of empty panels array, even when dashboard is newly created without any panels
+	// Due to this, we do not handle non "ok"
+	panels, ok := dashboardData["panels"]
+	if !ok {
+		return emptyResponse
+	}
+
+	panelList, ok := panels.([]interface{})
+	if !ok {
+		return emptyResponse
+	}
+
+	return panelList
+}
+
+func isAnalyticsPanel(panel map[string]interface{}) bool {
+	panelType, ok := panel["type"]
+	if ok {
+		return panelType == "macropower-analytics-panel"
+	}
+
+	return false
+}
+
+func (api *Client) getResponseBody(res *http.Response) []byte {
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "Failed to get response body",
+			"error", err,
+		)
+
+		return []byte{}
+	}
+
+	return body
+}
+
+func (api *Client) prepare(method string, endpoint string, payload []byte) (*http.Request, error) {
+	bearer := "Bearer " + api.Token
+
+	req, err := http.NewRequest(method, api.GrafanaUrl+endpoint, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, errors.New("Failed to create new HTTP Request")
```

Comment:
add more context to error message. Endpoint and maybe method could be useful to understand from the logs